fat: Fix incomplete comparing in compare_name and compare_short_name

yuuzi41 opened the pull request:
compare_name function and compare_short_name function compare incompletely if the argument "name" is shorter than a name of directory entries.
This causes a failure to load EFI as reported in #164

Signed-off-by: Yuji Hagiwara <yuuzi41@gmail.com>

rbradford commented:
Thanks for your contribution. 

Please commit using `-s` to give the correct sign-off. And use Fixes:#164 in the commit message.

yuuzi41 commented:
Thank you for your suggestion.
I corrected the signed-off and the commit message as you suggested.
And also I added one more commit to fix the CI failure due to an effect of this PR.

retrage reviewed (approved):
Thank you for the PR. It looks good. This change adds a more accurate FAT directory name comparison.
